### PR TITLE
Document `TryFromBytes`

### PR DESCRIPTION
**Pull request by jswrenn**
Makes progress towards #5.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1146,18 +1148,87 @@ pub unsafe trait NoCell {
         Self: Sized;
 }
 
-/// Types whose validity can be checked at runtime, allowing them to be
-/// conditionally converted from byte slices.
+/// Implements [`TryFromBytes`].
 ///
-/// WARNING: Do not implement this trait yourself! Instead, use
-/// `#[derive(TryFromBytes)]`.
+/// This derive synthesizes the runtime checks required to check if a sequence
```
Comment (joshlf) [inline src/lib.rs:1153]:
```suggestion
/// This derive synthesizes the runtime checks required to check whether a sequence
```

File: src/lib.rs [hunk 2]
```diff
@@ -1146,18 +1148,87 @@ pub unsafe trait NoCell {
...
+///
+/// [safety conditions]: trait@TryFromBytes#safety
+#[cfg(any(feature = "derive", test))]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "derive")))]
+pub use zerocopy_derive::TryFromBytes;
+
+/// Types for which some bit patterns are valid.
+///
+/// A memory region of the appropriate length which contains initialized bytes
+/// can be viewed as a `TryFromBytes` type so long as the runtime value of bytes
```
Comment (joshlf) [inline src/lib.rs:1192]:
```suggestion
/// can be viewed as a `TryFromBytes` type so long as the runtime value of those bytes
```

File: src/lib.rs [hunk 3]
```diff
@@ -1146,18 +1148,87 @@ pub unsafe trait NoCell {
...
+/// correspond to a [*valid instance*] of that type. For example, [`bool`] is
```
Comment (joshlf) [inline src/lib.rs:1193]:
```suggestion
/// corresponds to a [*valid instance*] of that type. For example, [`bool`] is
```

File: src/lib.rs [hunk 4]
```diff
@@ -1146,18 +1148,87 @@ pub unsafe trait NoCell {
...
+/// `TryFromBytes`; zerocopy can transmute a [`bool`] into a [`u8`] so long as
```
Comment (joshlf) [inline src/lib.rs:1194]:
```suggestion
/// `TryFromBytes`; zerocopy can transmute a [`u8`] into a  [`bool`] so long as
```

Also, this isn't technically true in our API (right now). We don't have `try_transmute!`, so we can't actually do this. Maybe say something like "zerocopy can transmute a `[u8]` into a `bool`..."?

**Review by joshlf — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (jswrenn) [reply src/lib.rs]: Done

File: src/lib.rs (hunk 2 above)
Comment (jswrenn) [reply src/lib.rs]: Done

File: src/lib.rs (hunk 4 above)
Comment (jswrenn) [reply src/lib.rs]: Done

**Review by joshlf — APPROVED** [review]

**Pull request merged**